src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -1254,6 +1280,13 @@ public void onAddTable() {
 			postCheckCreateTableOrView(TableType.table);
 		});
 	}
+	
+	public void onAddMaterializedView() {
+		preflightController.checkCreateEntity(entityBundle, MaterializedView.class.getName(), () -> {
+			// to create a MaterializedView, we need to know the definingSQL
+			getMaterializedViewEditor().configure(entityBundle.getEntity().getId()).show();
+		});
```

Comment:
A user could create a materialized view with null definingSql if they close this, so the UI needs to be able to handle that case.